src/engine/blocks.js

```diff
@@ -326,12 +327,15 @@ class Blocks {
             break;
         case 'checkbox':
             block.isMonitored = args.value;
+            block.targetId = args.isSpriteSpecific ? this._getTargetIdFromBlockId(block.id) : null;
```

Comment:
since checkbox events can only be triggered when on the current editing target, maybe we can simplify this and avoid the brittle "parse the target id out of the block id" thing by just using the current editing target?

Reply:
Great idea, thanks Paul!